converter.go

```diff
@@ -1634,96 +1637,306 @@ func buildListFromNativeArrow(ctx context.Context, rowIdx int, fieldMetadata fie
 		case *array.Decimal128:
 			if higherPrecision && fieldMetadata.Scale == 0 {
 				return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (*big.Int, error) {
-					return arrowDecimal128ToValue(typedValues, j, higherPrecision, fieldMetadata.Scale).(*big.Int), nil
+					bigInt := arrowDecimal128ToValue(typedValues, j, higherPrecision, fieldMetadata.Scale)
+					if bigInt == nil {
```

Comment:
Why do we need this change?

Reply:
Thats the same issue as we've discussed with the []byte. The assertion on nil results in an unnecessary panic.

Reply:
Ok, so I'm not getting it. This one won't work correctly IMHO. This function returns `snowflakeValue` which is an effective alias for `interface{}`. So here we never have `nil` - it's always "typed" nil, and this condition is never true... Try running `TestWithHigherPrecision` and stop there in debug mode :)

Reply:
I've added a test  `TestArraysWithNullValuesHigherPrecision` to demonstrate. Try running it now and with the previous code

Reply:
Ok, I see. This smells, because `arrowDecimal128ToValue` may return string... It's ok then.

Reply:
What does returning a string change in this context?

Reply:
If we didn't return string, we could return typed `*big.Int` instead of `any` - and this if wouldn't be needed at all.